### ISSUE TITLE
Sweep: Extend support for batch operation mocking / assertions

### ISSUE THREAD

**Issue by acline443** [pashagolub/pgxmock#149]
### Details

I may be misusing the library, so correct me if I am wrong. But I would like to write unit tests for functions that queue batches and then send the batch at the end. I would like to assert on the Updates/Insertions that are executed. 

It seems to me like this is not currently supported as the SendBatch method does basically nothing:
```go
func (c *pgxmock) SendBatch(context.Context, *pgx.Batch) pgx.BatchResults {
	return nil
}```

Best assertion I can do is assert that the queued batch length is greater than 0.

**Comment by pashagolub** [comment]
Yes, unfortunately, `SendBatch()` is not currently supported. I do have plans for it. However, this will require some spare weeks to implement. One needs to implement [pgx.BatchResults](https://pkg.go.dev/github.com/jackc/pgx/v5#BatchResults) from scratch and incorporate this into the expectations paradigm. Of course, pull requests are more than welcome!
